modules/custom/lua/dynamic_lottery_chance.lua

```diff
@@ -0,0 +1,232 @@
+--
+-- Because there are 9 placeholders, it will take 9 * 50 = 450 kills to reach the maximum lottery chance
+-- This rule says that on the first PH to be killed, the chance to spawn should be 5% * 0.8 = 4%.
+-- The maximum lottery chance after 450 kills will be 5% * 3.0 = 15%, which has increased linearly from 4% on every single kill
+-- Because 15% is less than 50%, the absoluteMaxValue is irrelevant here
+-- When Aquarius spawns, the kill counter is reset and the first placeholder killed has a 4% chance to spawn Aquarius again
+
+-- Example 2
+-- ["Wyvernpoacher_Drachlox"] = { minValue=1.0, maxValue=5.0, absoluteMaxValue=20, killsToMaxValue=500, useTimeInsteadOfKills=true }
+-- Drachlox normally has a 5% chance to spawn from 1 placeholder, which respawns every 16 minutes
+--
+-- Because this is a time-based rule, rather than 500 kills it will take 500 * 16m = 133 hours to reach maxValue
+-- The maxValue is reached 133 hours after the spawn window opens regardless of whether or not any of the placeholders are killed
+-- The spawn chance will be 5% * 1.0 = 5% if the placeholder is killed immediately when the window opens
+-- When the placeholder is killed after 133 hours, the spawn chance will be 5% * 5.0 = 25%
+-- Because absoluteMaxValue is 20, the actual spawn chance after 133 hours is only 20%, not 25%
+-- When Drachlox dies, the lottery chance is reset back to 5% once the 2-hour minimum respawn timer has expired
+
+--local defaultRule = nil
+local defaultRule = { minValue=0.8, maxValue=3.0, absoluteMaxValue=25, killsToMaxValue=50 }
+
+local overrideList = {
+    --["Wyvernpoacher_Drachlox"] = { maxValue=4.0, killsToMaxValue=500 }
+}
+
+local exclusionList = {
+    -- NMs with odd phLists that contain more values than just their placeholders which may cause unexpected behavior
+    "Leaping_Lizzy",
+    "Bloodtear_Baldurf", 
+    "Lumbering_Lambert",
+    "Steelfleece_Baldarich",
+    "Rampaging_Ram",
+    -- NMs that only appear at specific times of day will work with useTimeInsteadOfKills=false, but will give incorrect results if true
+    "Citipati",
+    -- NMs that have PHs with different respawn timers may not be 100% accurate, but not enough where it's a problem
+    --"Poisonhand_Gnadgad",
+    -- PHs with variable respawn timers may produce unintended results
+    "Fradubio",
+    -- NMs with multiple placeholders, but not all can be spawned at the same time
+    --"Charybdis"
+}
+
+local multiplyKillsToMaxByNumberOfPlaceholders = true
+
+-- TODO: Get rid of this, and instead call nm:getZone():getInitializedTime()
+-- Such a function doesn't exist yet, though
+-- If that isn't possible, technically this should override all zones and track them individually,
+-- but there doesn't seem to be a good way to iterate over them all. Also they'll be reset on hot-reload
+local zoneInitializedTime = 0
+m:addOverride("xi.zones.Qufim_Island.Zone.onInitialize", function(zone)
+    super(zone)
+    zoneInitializedTime = os.time()
```

Comment:
This won't work - you would need to set this as a server var to have it accessible to this module, also as stated in my main comment this only will pull the uptime for the instance that has qufim on it

Reply:
Yeah, @Frankie-hz mentioned he had some code for Horizon that allowed you to get uptime that this should ideally be using if he can submit it to ASB. Otherwise I might add in some code to do something similar.